### ISSUE TITLE
Lock on the temp set_env file in GitHub actions

### ISSUE THREAD

**Issue by glennawatson** [dotnet/Nerdbank.GitVersioning#561]
I been getting this issue consistently with GitHub actions for a little while now, it's intermittent so will not happen every every build.

Seems that the file hosting environment variables get locked a lot. See below for stack trace.

This is with version 3.3.37 of GitVersioning both on the CI build server/nuget packages.

If you need any more assistance on this one let me know.

One possible thing it may be we do run builds in multicpu mode with /m on msbuild. So maybe multiple builds happening at once trying to use the same file?

```
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.SetCloudBuildVariables" task failed unexpectedly. [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018: System.IO.IOException: The process cannot access the file 'D:\a\_temp\_runner_file_commands\set_env_9025d586-02c7-4780-a2e4-9faa48aa1f45' because it is being used by another process. [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.File.InternalAppendAllText(String path, String contents, Encoding encoding) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at System.IO.File.AppendAllText(String path, String contents) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at NerdBank.GitVersioning.CloudBuildServices.GitHubActions.SetCloudBuildVariable(String name, String value, TextWriter stdout, TextWriter stderr) [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.SetCloudBuildVariables.Execute() [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
C:\Users\runneradmin\.nuget\packages\nerdbank.gitversioning\3.3.37\build\Nerdbank.GitVersioning.targets(115,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI.Events.WPF\ReactiveUI.Events.WPF.csproj]
```

**Comment by glennawatson** [comment]
I've turned off the multicpu mode on msbuild to see if it helps the situation. Those GitHub Actions VM only have 2 CPUs anyway.

**Comment by AArnott** [comment]
Fixed by #556, available in 3.4.168-alpha.

**Comment by glennawatson** [comment]
Great thankyou.
